patches/tModLoader/Terraria.ModLoader.Core/ModCompile.cs

```diff
@@ -595,38 +595,14 @@ private class BuildingMod : LocalMod
 			}
 		}
 
-		private string DllRefPathForCompilation(BuildingMod mod, string dllName, bool xna)
-		{
-			string path = Path.Combine(mod.path, $"lib/{dllName}");
-			string refPath = DllRefPath(path, dllName, xna);
-			if (refPath != null)
-				return refPath;
-
-			throw new BuildException("Missing dll reference: " + path);
-		}
-
-		private IEnumerable<string> DllRefPathsForPackaging(BuildingMod mod, string dllName)
-		{
-			string path = Path.Combine(mod.path, $"lib/{dllName}");
-			string xnaRefPath = DllRefPath(path, dllName, true);
-			string fnaRefPath = DllRefPath(path, dllName, false);
-
-			string[] refPaths = new[] {xnaRefPath, fnaRefPath}.Where(p => p != null)
-				.Distinct().ToArray();
+		private string DllRefPath(BuildingMod mod, string dllName, bool? xna) {
+			var path = Path.Combine(mod.path, "lib", dllName + ".dll");
```

Comment:
The code below already appends the extension, so this results in file names like "DllName.dll.dll" or "DllName.dll.XNA.dll".